[Bug] Compilation error when enabling both Bluetooth and pointer device

<!-- Provide a general summary of the bug in the title above. -->

<!--- This template is entirely optional and can be removed, but is here to help both you and us. -->
<!--- Anything on lines wrapped in comments like these will not show up in the final text. -->

## Describe the Bug

<!-- A clear and concise description of what the bug is. -->

In `rules.mk`, with `BLUETOOTH_ENABLE = yes` and `POINTING_DEVICE_ENABLE = yes` (i.e. with both Bluetooth and pointer device enabled) e.g.
```mk
BLUETOOTH_ENABLE = yes
BLUETOOTH_DRIVER = AdafruitBLE # or RN42

POINTING_DEVICE_ENABLE = yes
POINTING_DEVICE_DRIVER = analog_joystick
```

will cause a compilation error:

```
Compiling: drivers/bluetooth/adafruit_ble.cpp                                                      In file included from quantum/quantum.h:216,
                 from platforms/avr/drivers/analog.h:20,
                 from drivers/bluetooth/adafruit_ble.cpp:13:
quantum/pointing_device.h:80:56: error: expected ',' or '...' before 'new'
 bool           has_mouse_report_changed(report_mouse_t new, report_mouse_t old);
                                                        ^~~
 [ERRORS]
 |
 |
 |
make[1]: *** [tmk_core/rules.mk:456: .build/obj_kb_default/drivers/bluetooth/adafruit_ble.o] Error 1
Make finished with errors
make: *** [Makefile:478: kb:default] Error 1
```


## System Information

 - Keyboard:  Unreleased keyboard in testing (Using Pro Micro)
 - Operating system: Windows 10
 - AVR GCC version: 8.4.0
<!-- Run `avr-gcc --version` to find this out. -->
 - ARM GCC version: 10.1.0
<!-- Run `arm-none-eabi-gcc --version` to find this out. -->
 - QMK Firmware version: 0.15.16
<!-- Run `git describe --abbrev=0 --tags` to find this out. -->
 - Any keyboard related software installed? none
   - [ ] AutoHotKey
   - [ ] Karabiner
   - [ ] Other:

## Additional Context

<!-- Add any other relevant information about the problem here. -->

The error refers to this file: [pointing_device.h](https://github.com/qmk/qmk_firmware/blob/ae5f43072f0de93b65854af569ba90564b8003de/quantum/pointing_device.h#L80)

I think the compiler is mistaking the parameter name "new" for "new operator", which causes this error.

I tried changing the name of the parameter from `new` to `newMouseReport`, and the above problem was solved. I think changing the parameter name to `newMouseReport` can also make it consistent with the other functions (`pointing_device_set_report(report_mouse_t newMouseReport)`).

--- a/drivers/sensors/analog_joystick.c
+++ b/drivers/sensors/analog_joystick.c
@@ -18,6 +18,8 @@
 #include "analog.h"
 #include "gpio.h"
 #include "wait.h"
+#include "timer.h"
+#include <stdlib.h>
 
 // Set Parameters
 uint16_t minAxisValue = ANALOG_JOYSTICK_AXIS_MIN;

--- a/platforms/avr/drivers/analog.c
+++ b/platforms/avr/drivers/analog.c
@@ -14,9 +14,6 @@
  * along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
 
-#include <avr/io.h>
-#include <avr/pgmspace.h>
-#include <stdint.h>
 #include "analog.h"
 
 static uint8_t aref = ADC_REF_POWER;

--- a/platforms/avr/drivers/analog.h
+++ b/platforms/avr/drivers/analog.h
@@ -17,7 +17,7 @@
 #pragma once
 
 #include <stdint.h>
-#include "quantum.h"
+#include "gpio.h"
 
 #ifdef __cplusplus
 extern "C" {

--- a/platforms/chibios/drivers/analog.c
+++ b/platforms/chibios/drivers/analog.c
@@ -14,7 +14,6 @@
  * along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
 
-#include "quantum.h"
 #include "analog.h"
 #include <ch.h>
 #include <hal.h>

--- a/platforms/chibios/drivers/analog.h
+++ b/platforms/chibios/drivers/analog.h
@@ -17,7 +17,7 @@
 #pragma once
 
 #include <stdint.h>
-#include "quantum.h"
+#include "gpio.h"
 
 #ifdef __cplusplus
 extern "C" {

